spec/mastermind_spec.rb

```diff
@@ -1,82 +1,101 @@
 require 'mastermind'
 
 
-game = Mastermind.new
+RSpec.describe Mastermind do
+	let(:game) { Mastermind.new }
+	let(:colors) { ['R' , 'O', 'Y', 'G', 'B', 'P'] }
 
-RSpec.describe Mastermind, "Print Directions" do
-
-	it "prints game play instructions to console" do
-		expect {game.print_directions}.to output(/To play:/).to_stdout 
+	describe '#colors' do
+		it 'returns the colors array' do
+			expect(game.colors).to eq (colors)
+		end
 	end
 
-end
-
-RSpec.describe SecretCode, "Generate" do 
-	game = Mastermind.new
-	it "generates a secret code that only includes valid colors" do
-		secret_code = SecretCode.new
-		code = secret_code.generate(game.colors)
-		colors = game.colors
-		pass = true
-
-		code.each do |element| 
-			if (!(colors.include?(element)))
-				pass = false
-			end
+	describe '#secret_code' do
+		it 'returns an empty array when initialized' do
+			expect(game.secret_code.class).to eq (Array)
 		end
-
-		expect(pass).to eq(true)
 	end
 
-	it " generates a secret code of 4 colors" do
-
-		secret_code = SecretCode.new
-		code = secret_code.generate(game.colors)
-		
-		expect(code.size).to eq 4
+	describe '#secret_code' do
+		it 'can be changed' do
```

Comment:
This feels like it's testing ruby functions (attr_accessor)...which isn't that useful most of the time